### PR TITLE
Issue #6696: Create appveyor batch file

### PR DESCRIPTION
**Pull request by ahmadshabib**
Issue #6696

### CODE REVIEW

File: .ci/appveyor.bat [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+mvn -e -Pno-validations site^
+ -Dtest=!FileContentsTest#testGetJavadocBefore,!FileTextTest#testFindLine*,^
+!MainFrameModelPowerTest#testOpenFileWithUnknownParseMode,^
+!TokenUtilTest#testTokenValueIncorrect2,^
+!ImportControlLoaderPowerTest#testInputStreamThatFailsOnClose^
+ -Djacoco.skip=true
+  goto END_CASE
+
+:DEFAULT_CASE
+  echo unknown option
```
Comment (rnveach) [inline .ci/appveyor.bat]: Can we print unexpected option like in travis.sh?

File: appveyor.yml [hunk 2]
```diff
@@ -61,89 +61,60 @@ matrix:
...
-      CMD1: "mvn -e verify -Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true"
+      CMD: "./.ci/appveyor.bat verify_without_checkstyle"
     # verify without checkstyle (JDK11)
     # Spotbugs check is disabled till https://github.com/spotbugs/spotbugs/issues/259
     - JAVA_HOME: C:\Program Files\Java\jdk11
       DESC: "verify without checkstyle (JDK11)"
-      CMD1: "mvn -e verify -Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true"
+      CMD: "./.ci/appveyor.bat verify_without_checkstyle_JDK11"
     # verify without checkstyle (JDK12)
     # Spotbugs check is disabled till https://github.com/spotbugs/spotbugs/issues/259
```
Comment (rnveach) [inline appveyor.yml]: Move spotbug comment to batch file.

File: appveyor.yml [hunk 3]
```diff
@@ -61,89 +61,60 @@ matrix:
...
       DESC: "verify without checkstyle (JDK9)"
-      CMD1: "mvn -e verify -Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true"
+      CMD: "./.ci/appveyor.bat verify_without_checkstyle"
     # verify without checkstyle (JDK10)
     - JAVA_HOME: C:\Program Files\Java\jdk10
       DESC: "verify without checkstyle (JDK10)"
-      CMD1: "mvn -e verify -Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true"
+      CMD: "./.ci/appveyor.bat verify_without_checkstyle"
     # verify without checkstyle (JDK11)
     # Spotbugs check is disabled till https://github.com/spotbugs/spotbugs/issues/259
```
Comment (rnveach) [inline appveyor.yml]: same.

File: .ci/appveyor.bat [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+:verify_without_checkstyle
+mvn -e verify -Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true
+  goto END_CASE
+
+:verify_without_checkstyle_JDK11
+mvn -e verify -Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true
+  goto END_CASE
+
+:verify_without_checkstyle_JDK12
+mvn -e verify -Dcheckstyle.ant.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true^
```
Comment (rnveach) [inline .ci/appveyor.bat]: Please add comment similar to travis.sh.
> `# powermock doesn't support modifying final fields in JDK12`

File: .ci/appveyor.bat [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ -Djacoco.skip=true
+  goto END_CASE
+
+:site_without_verify
+mvn -e -Pno-validations site
+  goto END_CASE
+
+:site_without_verify_jdk12
+mvn -e -Pno-validations site^
+ -Dtest=!FileContentsTest#testGetJavadocBefore,!FileTextTest#testFindLine*,^
```
Comment (rnveach) [inline .ci/appveyor.bat]: same.

File: .ci/appveyor.bat (hunk 1 above)
Comment (ahmadshabib) [reply .ci/appveyor.bat]: Yes, i added that

File: appveyor.yml (hunk 2 above)
Comment (ahmadshabib) [reply appveyor.yml]: Done

File: appveyor.yml (hunk 3 above)
Comment (ahmadshabib) [reply appveyor.yml]: Done.

File: .ci/appveyor.bat (hunk 5 above)
Comment (ahmadshabib) [reply .ci/appveyor.bat]: Done

File: .ci/appveyor.bat (hunk 4 above)
Comment (ahmadshabib) [reply .ci/appveyor.bat]: Done

**Comment by rnveach** [comment]
@ahmadshabib Please leave it to us to mark things as resolved so we can use to re-review the changes.

File: appveyor.yml [hunk 6]
```diff
@@ -1,153 +1,122 @@
-os: Windows Server 2012
```
Comment (rnveach) [inline appveyor.yml]: Did the line endings change in this file? Why is everything marked as deleted and then re-added?

Comment (ahmadshabib) [reply appveyor.yml]: Yeah, my ide is using CRLF in default i had to change it fo LF.

Comment (rnveach) [reply appveyor.yml]: @ahmadshabib are you using windows? if so you should turn autocrlf on as described in http://checkstyle.sourceforge.net/beginning_development.html#Before_development .

Comment (rnveach) [reply appveyor.yml]: @romani ping because of other conversations going on about `autocrlf`.

**Comment by rnveach** [comment]
@ahmadshabib spellchecker is still failing.

**Comment by romani** [comment]
@ahmadshabib ,

 from Appveyor:
> [INFO] ------------------------------------------------------------------------
3496The system cannot find the batch label specified - END_CASE
3497Command exited with code 1

please address.

File: appveyor.yml [hunk 7]
```diff
@@ -61,89 +61,58 @@ matrix:
...
-       + (get-content env:CMD2)
-       + (get-content env:CMD3)
-       + (get-content env:CMD4)
-       + (get-content env:CMD5)
-       + (get-content env:CMD6)
-       + (get-content env:CMD7)
-       + (get-content env:CMD8)
-       + (get-content env:CMD9)
-       + (get-content env:CMD10)
+       (get-content env:CMD)
```
Comment (romani) [inline appveyor.yml]: can we avoid cmd file creation and directly call required command ?
Something like this:
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-expression?view=powershell-6

**Comment by ahmadshabib** [comment]
> @ahmadshabib ,
> 
> from Appveyor:
> 
> > [INFO] ------------------------------------------------------------------------
> > 3496The system cannot find the batch label specified - END_CASE
> > 3497Command exited with code 1
> 
> please address.

I am fixing this today. the casue might be that i am using windows so far.

File: appveyor.yml (hunk 7 above)
Comment (ahmadshabib) [reply appveyor.yml]: Yes i will work on that too.

**Comment by romani** [comment]
@ahmadshabib , please finish PR.

**Comment by ahmadshabib** [comment]
> @ahmadshabib , please finish PR.

Sorry i was busy a bit.
Working on it

**Comment by romani** [comment]
@ahmadshabib , please find time to finish PR.

**Comment by romani** [comment]
@ahmadshabib , ping, it would be awesome to finish this PR.

**Comment by ahmadshabib** [comment]
> @ahmadshabib , ping, it would be awesome to finish this PR.

@romani  can i know the reason of the wercker build failing since i have no access there ?

**Comment by rnveach** [comment]
@ahmadshabib 
> Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to instantiate 'OrderedProperties' class

Please ensure you are rebased on the latest master.

**Pull request merged**